app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt

```diff
@@ -7,35 +7,61 @@ import ar.com.wolox.android.R
 import ar.com.wolox.android.example.model.News
 import ar.com.wolox.android.example.ui.example.signup.SignUpActivity
 import ar.com.wolox.wolmo.core.fragment.WolmoFragment
-import ar.com.wolox.wolmo.core.presenter.BasePresenter
 import kotlinx.android.synthetic.main.fragment_home_news.*
+import java.util.LinkedList
 import javax.inject.Inject
 
-class HomeNewsFragment @Inject constructor() : WolmoFragment<BasePresenter<Any>>() {
+class HomeNewsFragment @Inject constructor() : WolmoFragment<HomeNewsPresenter>(), IHomeNewsView {
+
+    private val newsViewAdapter = HomeNewsViewAdapter()
+    private val newsList = LinkedList<News>()
 
     override fun init() {
+        vHomeNewsSwipeRefreshLayout.setColorSchemeResources(R.color.colorWoloxGreen,
+                R.color.colorWoloxBlue)
+
         vHomeNewsRecyclerView.apply {
-            // TODO FOR NEXT PR (Backend): change this list for data from a request to DB/API
-            adapter = HomeNewsViewAdapter(listOf(News("Detuvieron a Alberto Samid en Belice",
-                    "En Belice, fue detenido por Interpol Alberto Samid. Nuestra" +
-                            " política es clara y transparente. Quienes tienen deudas con la" +
-                            " justicia, deben"), News("Susto para Lionel Scaloni en España:" +
-                    " fue atropellado por un auto y sufrió lesiones", "El DT de la " +
-                    "Argentina iba en bicicleta y fue embestido en el estacionamiento del " +
-                    "colegio de sus hijas. Fue atendido en un hospital y ya recibió el alta.")))
             layoutManager = LinearLayoutManager(context)
             addItemDecoration(DividerItemDecoration(context, DividerItemDecoration.VERTICAL))
+            adapter = newsViewAdapter
         }
+
+        presenter.onInit()
     }
 
     override fun setListeners() {
         vHomeNewsAddButton.setOnClickListener {
             val intent = Intent(context, SignUpActivity::class.java).apply {
                 flags = Intent.FLAG_ACTIVITY_NEW_TASK
             }
+
             startActivity(intent)
         }
+
+        vHomeNewsSwipeRefreshLayout.setOnRefreshListener {
+            presenter.onRefresh()
+        }
     }
 
     override fun layout(): Int = R.layout.fragment_home_news
+
+    override fun onNewsGotFromDatabase(news: List<News>?) {
+        if (news != null) {
+            for (n in news) {
+                newsList.push(n)
+            }
+
+            newsViewAdapter.submitList(LinkedList<News>(newsList))
+        }
+    }
+
+    override fun onRefresh(news: List<News>?) {
+        // This list is a hardcoded one, for test. Instead using the given from parameter
+        val newestNews = LinkedList<News>()
+        newestNews.add(News(69, 1, "2019-01-18T14:00:29.985Z", "Training",
```

Comment:
This onRefresh is sending a harcoded news, why it is implemented like this?

Reply:
The given API REST does not support GET newer news. The news given from parameter are the same that are in the list. So, I addes a hardcoded one to differentiate the original ones (already in the list) from the one I add when you pull down. 

We can pass **news** instead **newestNews** to the **onNewsGotFromDatabase(List<News>)** function, but for me it was not clear to see that on the GIF or for visual-testing. That's why I let the hardcoded one.